testbed/datareceivers/carbon.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datareceivers
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/testbed/testbed"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/carbonreceiver"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/carbonreceiver/protocol"
+)
+
+// CarbonDataReceiver implements Carbon format receiver.
+type CarbonDataReceiver struct {
```

Comment:
These are great, but perhaps introduce these new data receivers and senders in a separate PR to avoid mixing introduction of the stability test with them?

Reply:
Makes sense. Please take a look https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/320 
I'll rebase after